httpcore5/src/main/java/org/apache/hc/core5/util/TimeValue.java

```diff
@@ -371,13 +347,43 @@ public int toSecondsIntBound() {
         return asBoundInt(toSeconds());
     }
 
+    public Timeout toTimeout() {
+        return Timeout.of(duration, timeUnit);
+    }
+
     @Override
-    public String toString() {
-        return String.format(Locale.ROOT, "%,d %s", duration, timeUnit);
+    public int compareTo(final TimeValue other) {
+        if (other != null) {
+            final TimeUnit targetTimeUnit = min(other.getTimeUnit());
+            return Long.compare(convert(targetTimeUnit), other.convert(targetTimeUnit));
+        } else {
+            return -Integer.MAX_VALUE;
+        }
     }
 
-    public Timeout toTimeout() {
-        return Timeout.of(duration, timeUnit);
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj instanceof TimeValue) {
+            final TimeValue that = (TimeValue) obj;
+            return this.duration == that.duration && LangUtils.equals(this.timeUnit, that.timeUnit);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = LangUtils.HASH_SEED;
+        hash = LangUtils.hashCode(hash, duration);
+        hash = LangUtils.hashCode(hash, timeUnit);
+        return hash;
+    }
+
+    @Override
+    public String toString() {
+        return String.format(Locale.ROOT, "%,d %s", duration, timeUnit);
```

Comment:
Don't use the group sep unless you know it is a space. Otherwise this might confuse people dependeng on their native language.

Reply:
@michael-o I did not write this method but is not there a specific `Locale` passed as a parameter?

Reply:
It is, but I highly doubt that `ROOT` is much different than `en_US`. Therefore, output should be in neutral formatting.

Reply:
@michael-o This code existed before my PR and is unrelated to `Comparable`. Please raise a separate PR with the changes you are proposing.